### DISCUSSION TITLE
How to compare values in tags dynamically?

### DISCUSSION THREAD

**Discussion by cmorierroy** [statamic/cms#5194] [Q&A] [answered]
Hi - pretty new to statamic, let me know if I'm not using it properly.

I'm trying render html based on the value of something in a collection.
Basically, iterate through collection, if there is a match, render. I don't know the value that will match beforehand.
I have js code to obtain the value, but can't find a way to call it inside tags.
Is there a way to use javascript inside tags to dynamically obtain that value? Or some other way to do this?

e.g.
```
{{ locations }}
     {{ if name== "place" }} <-- I need this "place" string to be dynamically generated, I can't predict what it will be.
           <p>Display me</p>
      {{ /if }}
{{ /location}}
```
```
{{ locations }}
     {{ if name== getLocation() }} <-- This definitely doesn't work, but hopefully helps illustrate the point.
           <p>Display me</p>
      {{ /if }}
{{ /location}}
```

Thanks!

**Comment by jasonvarga** [comment, marked as answer]
You can't use JavaScript in Antlers tags because it's rendered before it gets sent to the browser. JavaScript doesn't have a chance to touch it.

I guess you could output everything through Antlers and use some logic in your Javascript to display the right one.

Using Alpine.js as an example here. This Antlers code...

```
{{ locations }}
  <p x-if="location == '{{ name }}'">{{ name }}</p>
{{ /locations }}
```

would render...

```
  <p x-if="location == 'one'">one</p>
  <p x-if="location == 'two'">two</p>
  <p x-if="location == 'three'">three</p>
```

Then Alpine would display the right one.

Although something tells me this is also not a great solution. Maybe more details about what you're doing would be helpful and you could get a different suggestion.

**Reply by cmorierroy** [reply]
I want to display location specific information (maintained on the control panel) given a location cookie (user selects their location - no login system). The real issue here is probably using the cookie instead of using statamic to store that somehow - but that would be another discussion.

I don't mind the option of pre-rendering everything first, then hiding irrelevant entries. 
Not nice for scaling, but works for me short-term.
Thanks!

**Reply by jasonvarga** [reply]
You can read cookies from the server side. 

There's no built-in tag to get the cookie, but it's pretty simple to make. Plop this in `app/Modifiers/IsMatchingLocation`:

```php
<?php

namespace App\Modifiers;

use Statamic\Modifiers\Modifier;

class IsMatchingLocation extends Modifier
{
    public function index($value)
    {
        return $value === request()->cookie('yourcookie');
    }
}
```

Then you can do:

```html
{{ locations }}
  {{ if name|is_matching_location }}
    <p>display me</p>
  {{ /if }}
{{ /locations }}
```

If you're setting the cookie through JavaScript, you'll need to disable encryption of it on the server side by adding the cookie name here: https://github.com/statamic/statamic/blob/71639fca6ddd7724fa207c09ee3b669fbc5ba3ad/app/Http/Middleware/EncryptCookies.php#L15

```php
protected $except = [
    'yourcookie',
];
```

(I would have used a simpler tag and condition to get the cookie but I just noticed a parser bug. Didn't want to get into it on this response though, so here's a solution that works anyway.)

**Reply by cmorierroy** [reply]
Looks like I don't have that app folder. I'm on 2.11.19 - guessing this was added in a newer version?

**Reply by jasonvarga** [reply]
Oh yeah that solution is for v3. You said you're new to Statamic so I didn't think you'd be using v2.
